src/main/kotlin/org/kamiblue/client/module/modules/render/Xray.kt

```diff
@@ -0,0 +1,33 @@
+package org.kamiblue.client.module.modules.render
+
+import net.minecraft.block.state.IBlockState
+import org.kamiblue.client.module.Category
+import org.kamiblue.client.module.Module
+import org.kamiblue.client.setting.settings.impl.collection.CollectionSetting
+
+internal object Xray : Module(
+    name = "Xray",
+    description = "Lets you see through blocks",
+    category = Category.RENDER
+) {
+    private val defaultVisibleList = linkedSetOf("minecraft:diamond_ore", "minecraft:iron_ore", "minecraft:gold_ore")
+
+    val showFluids by setting("Show Fluids", false)
+
+    val visibleList = setting(CollectionSetting("Visible List", defaultVisibleList, { false }))
+
+    @JvmStatic
+    fun shouldReplace(state: IBlockState): Boolean {
+        return isEnabled && !visibleList.contains(state.block.registryName.toString())
+    }
+
+    init {
+        onToggle {
+            mc.renderGlobal.loadRenderers()
+        }
```

Comment:
Does this need to be onToggle or can it be on only enabled?

Reply:
Otherwise it doesn't disable correctly and you keep xray on until the chunk changes.

Reply:
Otherwise you are stuck with xray on even after disabling until there is a change in the chunk